Tests/SwiftCloudantTests/CouchClientTest.swift

```
@@ -12,19 +12,171 @@ import XCTest
 
 class CouchClientTest: XCTestCase {
 
+    private class VCAPGenerator {
+
+        private var VCAPServices = ["cloudantNoSQLDB": []]
+
+        public func addService(name: String?, url: String?) {
+            var VCAPService = [String:Any]()
+            if name != nil {
+                VCAPService["name"] = name!
+            }
+            if url != nil {
+                if url == "" {
+                    VCAPService["credentials"] = [] // empty credentials array
+                } else {
+                    VCAPService["credentials"] = ["url": url!]
+                }
+            }
+
+            VCAPServices["cloudantNoSQLDB"]!.append(VCAPService)
+        }
+
+        public func toJSONString() -> String? {
+            if let json = try? JSONSerialization.data(withJSONObject: VCAPServices,
```

since throwing will probably never happen since its test code, lets just let the method throw the error back up to the caller.